solution1.py

```diff
@@ -0,0 +1,29 @@
+# https://leetcode.com/problems/subarray-sum-equals-k
+# // Time Complexity : o(n)
+# // Space Complexity : o(n)
+# // Did this code successfully run on Leetcode : yes
+# // Any problem you faced while coding this : got confused in case k==0, realized I need to increase count first then add the new element in hash
+#
+
+class Solution:
+    def subarraySum(self, nums: List[int], k: int) -> int:
+        hm = dict()
+        hm[0] = 1
+        sum1 = []
+        seed = 0
+        count = 0
+
+        for num in nums:
+            temp = seed + num
+            sum1.append(temp)
+            seed = temp
+        for i in range(len(sum1)):
```

Comment:
Please try the solution with a single hashmap and no other Data Structure as well.